cmake/FindPugiXML.cmake

```diff
@@ -1,7 +1,22 @@
-find_path(PUGIXML_INCLUDE_DIR NAMES pugixml.hpp)
-find_library(PUGIXML_LIBRARIES NAMES pugixml)
+if(APPLE)
+
+	find_package(PkgConfig REQUIRED)
+	pkg_check_modules(PC_PUGIXML QUIET pugixml)
+	set(PUGIXML_DEFINITIONS ${PC_PUGIXML_CFLAGS_OTHER})
```

Comment:
Try using `add_compile_options(${PC_PUGIXML_CFLAGS_OTHER})` to apply flags directly.